Sources/Commands/SwiftPackageTool.swift

```diff
@@ -151,11 +151,19 @@ public class SwiftPackageTool: SwiftTool<PackageToolOptions> {
 
         case .update:
             let workspace = try getActiveWorkspace()
-            try workspace.updateDependencies(
-                root: getWorkspaceRoot(),
-                diagnostics: diagnostics
-            )
-
+            
+            if !options.updateDryRun {
+                try workspace.updateDependencies(
+                    root: getWorkspaceRoot(),
+                    diagnostics: diagnostics
+                )
+            } else if let pinsStore = diagnostics.wrap({ try workspace.pinsStore.load() }),
+                let changes = try workspace.updateDependenciesDryRun(
+                    root: getWorkspaceRoot(),
+                    diagnostics: diagnostics) {
```

Comment:
Didn't we discuss passing an inout array to get the changes? Any particular reason for switching to a new method?

Reply:
Since I had to create a new method due to API stability anyway, I chose to use a return value instead of `inout` because I think the **in** part of `inout` would never be used.

Reply:
I was thinking the parameter would default to nil and we'd the `in` part to determine if this is a dry run operation. Something like:

```swift
public func updateDependencies(
  root: PackageGraphRootInput,
  dryRunResults: [Result]? = nil
  diagnostics: DiagnosticsEngine
) {
...

  // Fill the dry run results array if one is provided and return without performing the actual update.
  if dryRunResults != nil {
    dryRunResults = updateResults
    return
  }
...
}
```

Reply:
This won't work in swift 5.1 inout parameters can not have a default value

Reply:
Ah. What if we add a specialized method ourselves?

```swift
public func updateDependencies(
  root: PackageGraphRootInput,
  diagnostics: DiagnosticsEngine
) {
  updateDependencies(root: root, dryRunResults: nil, diagnostics: diagnostics)
}

public func updateDependencies(
  root: PackageGraphRootInput,
  dryRunResults: [Result]?,
  diagnostics: DiagnosticsEngine
) {
...
}
```

Reply:
Is it obvious to someone reading the code for the first time that passing a dry run array causes the actual update not to happen?

Without reading the documentation it is more obvious that the method `updateDependencies` performs an actual dry run and `updateDependenciesDryRun` does not.
```
 updateDependencies(root: PackageGraphRootInput,
  dryRunResults: [Result]?,
  diagnostics: DiagnosticsEngine)
```

Reply:
You're right, it's not very obvious especially for API consumers. What if we always return the update result and make this method `@discardableResult`? Then this method could just take a simple Bool `dryRun`.